lib/src/mapbox_map.dart

```diff
@@ -505,6 +552,50 @@ class MapboxMap extends ChangeNotifier {
       _mapInterface.removeFeatureState(
           sourceId, sourceLayerId, featureId, stateKey);
 
+  /// Removes entries from a feature state object of a feature in the specified featureset.
+  /// Remove a specified property or all property from a feature's state object, depending on the value of `stateKey`.
+  @experimental
+  Future<void> removeFeatureStateForFeaturesetDescriptor(
+          {required FeaturesetDescriptor featureset,
+          required FeaturesetFeatureId featureId,
+          String? stateKey}) =>
+      _mapInterface.removeFeatureStateForFeaturesetDescriptor(
+          featureset, featureId, stateKey);
+
+  /// Removes entries from a specified Feature.
+  /// Remove a specified property or all property from a feature's state object, depending on the value of `stateKey`.
+  @experimental
+  Future<void> removeFeatureStateForFeaturesetFeature(
+          {required FeaturesetFeature feature, String? stateKey}) =>
+      _mapInterface.removeFeatureStateForFeaturesetFeature(feature, stateKey);
+
+  /// Reset all the feature states within a featureset.
+  ///
+  /// Note that updates to feature state are asynchronous, so changes made by this method might not be
+  /// immediately visible using ``MapboxMap/getFeatureState(_:callback:)``.
+  @experimental
+  Future<void> resetFeatureStatesForFeatureset(
+          FeaturesetDescriptor featureset) =>
+      _mapInterface.resetFeatureStatesForFeatureset(featureset);
+
+  // References for all interactions added to the map.
+  @experimental
+  var interactionsList = _InteractionsList(interactions: {});
+
+  /// Add an interaction
+  @experimental
+  void addInteraction(Interaction interaction, OnInteraction action) {
```

Comment:
@evil159  -- I'd appreciate a particularly thorough review of this section (setting up the listener and message channel. I model it broadly after our approach to gestures, but would welcome any suggested improvements. What do you think of this system of using both `_InteractionListener` and `_InteractionsList`?

Reply:
Given the limitations - I think this is perfectly fine way to do it(it is also probably the only way to do it :)

Reply:
I think the only thing that is missing here is the ability to remove interaction.